common/flatpak-dir.c

```diff
@@ -2147,98 +2185,178 @@ flatpak_deploy_data_get_installed_size (GVariant *deploy_data)
+  GLNX_HASH_TABLE_FOREACH_KV (values, const char *, locale, const char *, value)
     {
-      if (g_error_matches (my_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
-        {
-          data = g_strdup ("");
-        }
+      const char *key;
+      g_autofree char *key_free = NULL;
+      if (strcmp (locale, "C") == 0)
+        key = keyname;
       else
         {
-          g_propagate_error (error, g_steal_pointer (&my_error));
-          return NULL;
+          key_free = g_strdup_printf ("%s@%s", keyname, locale);
+          key = key_free;
         }
+
+      g_variant_builder_add (metadata_builder, "{s@v}", key,
+                             g_variant_new_variant (g_variant_new_string (value)));
     }
+}
 
-  lines = g_strsplit (data, "\n", 0);
+static void
+add_appdata_to_deploy_data (GVariantBuilder *metadata_builder,
+                            GFile      *deploy_dir,
+                            const char *id)
+{
+  g_autoptr(AsStore) store = as_store_new ();
+  g_autofree char *appdata_xml = NULL;
+  GPtrArray *apps;
 
-  subpaths = g_ptr_array_new ();
-  for (i = 0; lines[i] != NULL; i++)
+  appdata_xml = read_appdata_xml_from_deploy_dir (deploy_dir, id);
+  if (appdata_xml == NULL)
+    return;
+
+  as_store_from_xml (store, appdata_xml, NULL, NULL);
+  apps = as_store_get_apps (store);
+  if (apps->len > 0)
     {
-      lines[i] = g_strstrip (lines[i]);
-      if (lines[i][0] == '/')
-        g_ptr_array_add (subpaths, g_strdup (lines[i]));
+      AsApp *app = g_ptr_array_index (apps, 0);
+      AsRelease *release = as_app_get_release_default (app);
+
+      add_locale_metadata_string (metadata_builder, "appdata-name", as_app_get_names (app));
+      add_locale_metadata_string (metadata_builder, "appdata-summary", as_app_get_comments (app));
+      if (release)
+        g_variant_builder_add (metadata_builder, "{s@v}", "appdata-version",
+                               g_variant_new_variant (g_variant_new_string (as_release_get_version (release))));
     }
+}
+
+static GVariant *
+flatpak_dir_new_deploy_data (FlatpakDir *self,
+                             GFile      *deploy_dir,
+                             GVariant   *commit_metadata,
+                             GKeyFile   *metadata,
+                             const char *id,
+                             const char *origin,
+                             const char *commit,
+                             char      **subpaths,
+                             guint64     installed_size)
+{
+  char *empty_subpaths[] = {NULL};
+  GVariantBuilder metadata_builder;
+  g_autofree char *application_runtime = NULL;
+  const char *alt_id = NULL;
+  const char *eol = NULL;
+  const char *eol_rebase = NULL;
+
+  g_variant_lookup (commit_metadata, "xa.alt-id", "&s", &alt_id);
+  g_variant_lookup (commit_metadata, OSTREE_COMMIT_META_KEY_ENDOFLIFE, "&s", &eol);
+  g_variant_lookup (commit_metadata, OSTREE_COMMIT_META_KEY_ENDOFLIFE_REBASE, "&s", &eol_rebase);
```

Comment:
Now that we're growing support for localized strings here, it might be nice to support that for the eol strings too, eventually.